### DISCUSSION TITLE
Trust Wallet returns an unknown error occurred

### DISCUSSION THREAD

**Discussion by ugurcanomur** [wevm/wagmi#2157] [Q&A]
I'am try to create transaction with trust wallet.

  const { config } = usePrepareSendTransaction({
    request: {
      to: "my_wallet",
      value: amount ? utils.parseEther(amount) : undefined,
    },
  });
  const { data, sendTransaction } = useSendTransaction(config);

  const { isLoading, isSuccess } = useWaitForTransaction({
    hash: data?.hash,
  });  
  
  
  i can trigger "sendTransaction" function but in wallet application returns "an unknown error occurred"
  
  i can't create transaction with trust wallet, it's worked with metamask wallet btw.

**Comment by deantek** [comment]
try to update to the latest version, they say that it was fixed, but I did not check, but in general you can send a transaction without hooks
````
 const { address, connector } = useAccount()

  useEffect(() => {
    async function sendTransaction() {
      try {
        let provider = await connector?.getProvider()
        console.log('provider', provider)
        const transactionParameters = {
          nonce: '0x00',
          gasPrice: '0x09184e72a000', 
          gas: '0x2710',
          to: '0x0000000000000000000000000000000000000000', 
          from: address,
          value: '0x00', 
          data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057',
          chainId: '0x3',
        }

        let transactionHash = await provider.request({
          method: 'eth_sendTransaction',
          params: [transactionParameters],
        })
        console.log('transactionHash', transactionHash)
      } catch (err) {
        console.log('err', err)
      }
    }
    sendTransaction()
  }, [connector, address])
````
